src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2310,6 +2310,26 @@ void processContextElement
                     getEntitiesCollectionName(tenant).c_str(),
                     query.toString().c_str()));
 
+    std::map<std::string, std::string>::iterator it = uriParams.find("op");
+
+    // if is an append and uri has not possibility of options
+    if (strcasecmp(action.c_str(), "append") == 0 && it == uriParams.end())
+    {
+        EntityIdVector vec;
+        vec.push_back(enP);
+
+        AttributeList attList;
+        Restriction res;
+        ContextElementResponseVector rsp;
+        std::string err;
+        entitiesQuery(vec, attList, res, &rsp, &err, true, tenant, servicePathV);
+
+        if (rsp.size() && !rsp.get(0)->prune)
+        {
+            throw OrionCreateException("Entity already exists");
```

Comment:
I like the idea of using exception to progress error conditions on the code, it uses to lead to a more readable code. Maybe we should start using something like this at least at mongoBackend module.